src/signed-api-fetch/data-fetcher.ts

```diff
@@ -0,0 +1,200 @@
+
+// Useful for tests
+let axiosProd = axios;
+export const setAxios = (customAxios: any) => {
+  axiosProd = customAxios;
+};
+
+/**
+ * Shuts down intervals
+ */
+export const stopDataFetcher = () => {
+  clearInterval(getState().dataFetcherInterval);
+};
+
+/**
+ * Calls a remote signed data URL and inserts the result into the datastore
+ * @param url
+ * @param whoAmI
+ */
+const callSignedDataApi = async (url: string, whoAmI = 'unset'): Promise<SignedData[]> => {
+  const result = await go(
+    () =>
+      axiosProd({
+        method: 'get',
+        timeout: HTTP_SIGNED_DATA_API_ATTEMPT_TIMEOUT - HTTP_SIGNED_DATA_API_HEADROOM / 2,
+        url: url.indexOf('c52E') ? url : `https://pool.nodary.io/0xC04575A2773Da9Cd23853A69694e02111b2c4182`, // TODO ignore url for testing
+        headers: {
+          Accept: 'application/json',
+          // TODO add API key?
+        },
+      }),
+    {
+      attemptTimeoutMs: HTTP_SIGNED_DATA_API_ATTEMPT_TIMEOUT,
+      totalTimeoutMs: HTTP_SIGNED_DATA_API_ATTEMPT_TIMEOUT + HTTP_SIGNED_DATA_API_HEADROOM / 2,
+      retries: 0,
+    }
+  );
+
+  if (!result.success) {
+    throw new Error(`[${whoAmI}] HTTP call failed: ${result.error}`);
+  }
+
+  if (result.data.status !== 200) {
+    throw new Error(`[${whoAmI}] HTTP call failed with code ${result.data.status}: ${result.data.statusText}`);
+  }
+
+  const zodResult = signedApiResponseSchema.safeParse(result.data.data);
+  if (!zodResult.success) {
+    throw new Error(`[${whoAmI}] Schema parse failed with error(s) ${JSON.stringify(zodResult.error, null, 2)}`);
+  }
+
+  const payload = Object.values(zodResult.data.data);
+
+  if (!payload) {
+    throw new Error('Empty payload.');
+  }
+
+  return payload;
+};
+
+export const runDataFetcher = async () => {
+  const state = getState();
+  const { config } = state;
+
+  const fetchInterval = config.fetchInterval * 1_000;
+
+  if (!state?.dataFetcherInterval) {
+    const dataFetcherInterval = setInterval(runDataFetcher, fetchInterval);
+    setState({ ...state, dataFetcherInterval });
+  }
+
+  const urls = uniq(
+    Object.values(config.chains).flatMap((chain) =>
+      Object.entries(chain.__Temporary__DapiDataRegistry.airnodeToSignedApiUrl).flatMap(
+        ([airnodeAddress, baseUrl]) => `${baseUrl}/${airnodeAddress}`
+      )
+    )
+  );
+
+  return Promise.allSettled(
+    urls.map((url, idx) =>
+      go(
+        async () => {
+          const payload = await callSignedDataApi(url, idx.toString());
+
+          logErrors(await Promise.allSettled(payload.map(localDataStore.setStoreDataPoint)));
+        },
+        {
+          retries: 0,
+          totalTimeoutMs: fetchInterval + HTTP_SIGNED_DATA_API_HEADROOM,
+          attemptTimeoutMs: fetchInterval + HTTP_SIGNED_DATA_API_HEADROOM - 100,
+        }
+      )
+    )
+  );
+};
+
+// Everything from this point won't be needed in production.
```

Comment:
If you want to leave it in the codebase, I think you should move it to `test` folder.